docs/ja/docs/async.md

```diff
@@ -0,0 +1,393 @@
+
+```Python
+results = await some_library()
+```
+
+以下の様に `async def`を使用して*path operation 関数*を宣言します。
+
+```Python hl_lines="2"
+@app.get('/')
+async def read_results():
+    results = await some_library()
+    return results
+```
+
+!!! note "備考"
+    `async def` を使用して作成された関数の内部でしか `await` は使用できません。
+
+---
+
+データベース、API、ファイルシステムなどと通信し、`await` の使用をサポートしていない (現在のほとんどのデータベースライブラリに当てはまります) サードパーティを使用している場合、次の様に、単に `def` を使用して通常通り *path operation 関数* を宣言してください:
+
+```Python hl_lines="2"
+@app.get('/')
+def results():
+    results = some_library()
+    return results
+```
+
+---
+
+アプリケーションが (どういうわけか) 他の何とも通信せず、応答を待つ必要がない場合は、`async def` を使用して下さい。
+
+---
+
+よく分からない場合は、通常の `def` を使用して下さい。
+
+---
+
+**備考**: *path operation 関数*に必要なだけ `def` と `async def` を混在させ、それぞれに最適なオプションを使用して定義できます。FastAPIは正しい処理を行います。
+
+とにかく、上記のいずれの場合でもFastAPIは非同期で動作し、非常に高速です。
+
+しかし、上記のステップに従うことで、パフォーマンスの最適化を行えます。
+
+## 技術詳細
+
+現代版のPythonは「**非同期コード**」を、「**コルーチン**」と呼ばれるものと、**`async` と `await`** 構文でサポートしています。 
+
+次のセクションで、フレーズ内のパーツを順に見ていきましょう:
+
+* **非同期コード**
+* **`async` と `await`**
+* **コルーチン**
+
+## 非同期コード
+
+非同期コードとは、言語💬がコンピュータ/プログラム🤖にコード内のどこかの箇所でそれ🤖がどこかの *何か* が終了するのを待たなければならないと伝える手段を持っていることを意味します。*何か* は「遅いファイル📝」と呼ばれているとしましょう.
+
+したがって、コンピュータは「遅いファイル📝」が終了するまで、他の処理ができます。
+
+コンピュータ/プログラム🤖は再び待機する機会があるときや、その時点でそれ🤖が行っていたすべての作業が完了するたびに戻ってきます。そして、それ🤖が待っていた処理が終わっているか否か、確認します。
+
+次に、それ🤖が最初のタスク (「遅いファイル📝」としましょう) を受け付けて、それを使う必要があったことを続けます。
+
+この「何かを待つ」とは、通常、以下の様なものを待つ、相対的に「遅い」(プロセッサとRAMメモリの速度に比べて)<abbr title="インプットとアウトプット">I/O</abbr> 操作を指します:
+
+* ネットワーク経由でクライアントから送信されるデータ
+* ネットワーク経由でクライアントが受信する、プログラムから送信されたデータ
+* システムによって読み取られ、プログラムに渡されるディスク内のファイル内容
+* プログラムがシステムに渡して、ディスクに書き込む内容
+* リモートAPI操作
+* データベース操作の完了
+* 結果を返すデータベースクエリ
+* など。
+
+実行時間のほとんどが<abbr title="インプットとアウトプット">I/O</abbr> 操作の待ち時間が占めるため、このような操作を「I/O バウンド」操作と言います。
+
+コンピュータ/プログラムがこのような遅いタスクと「同期 (タスクの結果を取得して作業を続行するために、何もせずに、タスクが完了する瞬間を正確に待つ)」する必要がないため、「非同期」と呼ばれます。
+
+その代わりに、「非同期」システムであることにより、いったん終了すると、タスクは、コンピュータ/プログラムが既に開始した処理がすべて完了するのを少し (数マイクロ秒) 待って、結果を受け取りに戻ってきます。そして、処理を継続します。
+
+「同期」の場合 (「非同期」とは異なり)、「シーケンシャル」という用語もよく使用されます。これは、コンピュータ/プログラムがすべてのステップを (待機が伴う場合でも別のタスクに切り替えることなく) 順番に実行するためです。
+
+### 並行処理とハンバーガー
+
+上記の**非同期**コードのアイデアは、**並行処理**と呼ばれることもあります。 **並列処理**とは異なります。
+
+**並行処理**と**並列処理**はどちらも「多かれ少なかれ同時に発生するさまざまなこと」に関連しています。
+
+ただし、*並行処理*と*並列処理*の詳細はまったく異なります。
+
+違いを確認するには、ハンバーガーに関する次の物語を想像してみてください:
+
+### 並行ハンバーガー
+
+ファストフード🍔を食べようと、好きな人😍とレジに並んでおり、レジ係💁があなたの前にいる人達の注文を受けつけています。
+
+それからあなたの番になり、好きな人😍と自分のために、2つの非常に豪華なハンバーガー🍔を注文します。
+
+料金を支払います💸。
+
+レジ係💁はキッチンの男👨‍🍳に何か言ったので、その男はあなたのハンバーガー🍔を準備しなければならないと知っています (彼は現在、前のお客さんの商品を準備していますが)。
+
+レジ係💁はあなたに番号札を渡します。
+
+待っている間、好きな人😍と一緒にテーブルを選びに行き、好きな人😍と長い間座って話します (注文したハンバーガーは非常に豪華で、準備に少し時間がかかるので✨🍔✨)。
+
+好きな人😍とテーブルに座っているとき、ハンバーガー🍔を待つ間、あなたの好きな人がどれほど素晴らしく、かわいくてスマートであるか✨😍✨を賞賛しながら時間を費やすことができます。
+
+好きな人😍と話しながら待っている間、ときどき、カウンターに表示されている番号をチェックして、自分の番かどうかを確認します。
+
+その後、ついにあなたの番になりました。カウンターに行き、ハンバーガー🍔を手に入れてテーブルに戻ります。
+
+あなたとあなたの好きな人😍はハンバーガー🍔を食べて、楽しい時間を過ごします✨。
+
+---
+
+あなたがこのストーリーのコンピュータ/プログラム🤖であると想像してみてください。
```

Comment:
why past sentence?

Reply:
```suggestion
上記のストーリーで、あなたがコンピュータ/プログラム🤖だと想像してみてください。
```

"past" was not my intention. Wanted to tweak since at first I didn't catch that "このストーリーのコンピュータ/プログラム" points to the previous paragraph's guy. Let me suggest to emphasize the story comes from the last paragraph, and remove the past tense nuance.